File: labs/lab4/C#/Segment.cs
namespace LineLibrary
{
    public class Segment : Line
    {
        // Конструктор за замовчуванням
        public Segment() : base() { }

        // Конструктор із 3 параметрами (x, y, angle)
        // Явно викликаємо базовий конструктор на 4 параметри
        public Segment(double x, double y, double angle)
            : base(x, y,
                   x + System.Math.Cos(angle),
                   y + System.Math.Sin(angle))
        {
        }

        // Конструктор копіювання
        public Segment(Segment other) : base(other) { }

        // Метод обчислення кута з віссю OX
        public double AngleWithOx()
        {
            double dx = x2 - x1;
            double dy = y2 - y1;
            return System.Math.Atan2(dy, dx);
        }
    }
}

File: labs/lab4/C++/Segment.cpp
#include "Segment.h"
#include <cmath>

// За замовчуванням просто викликаємо базовий конструктор
Segment::Segment() : Line() {}

// Конструктор із 3 параметрами, де явно викликаємо конструктор базового класу
Segment::Segment(double x, double y, double angle)
    : Line(x, y, x + std::cos(angle), y + std::sin(angle))
{
    // Тут ми явно викликаємо конструктор Line(...) з 4 параметрами
}

// Конструктор копіювання
Segment::Segment(const Segment &other)
    : Line(other)  // виклик базового копіюючого
{}

// Кут з віссю OX (в радіанах)
double Segment::angleWithOx() const {
    double dx = x2 - x1;
    double dy = y2 - y1;
    return std::atan2(dy, dx);
}
